File: ios/ReactNativeCharts/combine/CombinedDataExtract.swift
//  Created by xudong wu on 02/03/2017.
//  Copyright © 2017 wuxudong. All rights reserved.
//

import Foundation

import SwiftyJSON
import Charts

class CombinedDataExtract : DataExtract {
    let lineDataExtract = LineDataExtract()
    let barDataExtract = BarDataExtract()
    let scatterDataExtract = ScatterDataExtract()
    let candleDataExtract = CandleDataExtract()
    let bubbleDataExtract = BubbleDataExtract()

    override func extract(_ data: JSON) -> ChartData?  {
        let chartData = CombinedChartData();


        if data["lineData"] != JSON.null {
            if let lineData = lineDataExtract.extract(data["lineData"]) as? LineChartData {
                chartData.lineData = lineData
            }
        }

        if data["barData"] != JSON.null {
            if let barData = barDataExtract.extract(data["barData"]) as? BarChartData {
                chartData.barData = barData
            }
        }


        if data["scatterData"] != JSON.null {
            if let scatterData = scatterDataExtract.extract(data["scatterData"]) as? ScatterChartData {
                chartData.scatterData = scatterData
            }
        }


        if data["candleData"] != JSON.null {
            if let candleData = candleDataExtract.extract(data["candleData"]) as? CandleChartData {
                chartData.candleData = candleData
            }
        }


        if data["bubbleData"] != JSON.null {
            if let bubbleData = bubbleDataExtract.extract(data["bubbleData"]) as? BubbleChartData {
                chartData.bubbleData = bubbleData
            }
        }
        return chartData

    }

    override func createEntries(_ values: [JSON]) -> [ChartDataEntry] {
        fatalError("not support")
    }


    override func createData() -> ChartData {
        fatalError("not support")
    }

    override func createDataSet(_ entries: [ChartDataEntry], label: String) -> ChartDataSetProtocol {
        fatalError("not support")
    }

    override func dataSetConfig(_ dataSet: ChartDataSetProtocol, config: JSON) {
        fatalError("not support")
    }

    override func createEntry(_ values: [JSON], index: Int) -> ChartDataEntry {
        fatalError("not support")
    }




}

File: android/src/main/java/com/github/wuxudong/rncharts/data/CombinedDataExtract.java
package com.github.wuxudong.rncharts.data;

import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.ReadableType;
import com.github.mikephil.charting.charts.Chart;
import com.github.mikephil.charting.data.CombinedData;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.interfaces.datasets.IDataSet;
import com.github.wuxudong.rncharts.utils.BridgeUtils;

import java.util.ArrayList;

/**
 * Created by xudong on 02/03/2017.
 */
public class CombinedDataExtract extends DataExtract<CombinedData, Entry> {
    private LineDataExtract lineDataExtract = new LineDataExtract();
    private BarDataExtract barDataExtract = new BarDataExtract();
    private ScatterDataExtract scatterDataExtract = new ScatterDataExtract();
    private CandleDataExtract candleDataExtract = new CandleDataExtract();
    private BubbleDataExtract bubbleDataExtract = new BubbleDataExtract();

    @Override
    public CombinedData extract(Chart chart, ReadableMap propMap) {
        CombinedData chartData = new CombinedData();

        if (BridgeUtils.validate(propMap, ReadableType.Map, "lineData")) {
            chartData.setData(lineDataExtract.extract(chart, propMap.getMap("lineData")));
        }

        if (BridgeUtils.validate(propMap, ReadableType.Map, "barData")) {
            chartData.setData(barDataExtract.extract(chart, propMap.getMap("barData")));
        }

        if (BridgeUtils.validate(propMap, ReadableType.Map, "scatterData")) {
            chartData.setData(scatterDataExtract.extract(chart, propMap.getMap("scatterData")));
        }

        if (BridgeUtils.validate(propMap, ReadableType.Map, "candleData")) {
            chartData.setData(candleDataExtract.extract(chart, propMap.getMap("candleData")));
        }

        if (BridgeUtils.validate(propMap, ReadableType.Map, "bubbleData")) {
            chartData.setData(bubbleDataExtract.extract(chart, propMap.getMap("bubbleData")));
        }

        return chartData;
    }

    @Override
    CombinedData createData() {
        throw new UnsupportedOperationException();
    }

    @Override
    public IDataSet<Entry> createDataSet(ArrayList<Entry> entries, String label) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void dataSetConfig(Chart chart, IDataSet<Entry> dataSet, ReadableMap config) {
        throw new UnsupportedOperationException();
    }

    @Override
    Entry createEntry(ReadableArray values, int index) {
        throw new UnsupportedOperationException();
    }
}
